tracing-appender/src/compression.rs

```diff
@@ -0,0 +1,154 @@
+use flate2::Compression;
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) enum GzipCompressionLevelLiteral {
+    None,
+    Fast,
+    Best,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) enum GzipCompressionLevelNumerical {
+    Level0,
+    Level1,
+    Level2,
+    Level3,
+    Level4,
+    Level5,
+    Level6,
+    Level7,
+    Level8,
+    Level9,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) enum GzipCompressionLevel {
+    Literal(GzipCompressionLevelLiteral),
+    Numerical(GzipCompressionLevelNumerical),
+}
+
+/// Defines a conversion between `CompressionOption` and `flate2::Compression`
+impl Into<Compression> for GzipCompressionLevel {
+    fn into(self) -> Compression {
+        match self {
+            GzipCompressionLevel::Literal(lit) => match lit {
+                GzipCompressionLevelLiteral::None => Compression::none(),
+                GzipCompressionLevelLiteral::Fast => Compression::fast(),
+                GzipCompressionLevelLiteral::Best => Compression::best(),
+            },
+            GzipCompressionLevel::Numerical(num) => match num {
+                GzipCompressionLevelNumerical::Level0 => Compression::new(0),
+                GzipCompressionLevelNumerical::Level1 => Compression::new(1),
+                GzipCompressionLevelNumerical::Level2 => Compression::new(2),
+                GzipCompressionLevelNumerical::Level3 => Compression::new(3),
+                GzipCompressionLevelNumerical::Level4 => Compression::new(4),
+                GzipCompressionLevelNumerical::Level5 => Compression::new(5),
+                GzipCompressionLevelNumerical::Level6 => Compression::new(6),
+                GzipCompressionLevelNumerical::Level7 => Compression::new(7),
+                GzipCompressionLevelNumerical::Level8 => Compression::new(8),
+                GzipCompressionLevelNumerical::Level9 => Compression::new(9),
+            },
+        }
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) struct GzipCompression {
+    pub(crate) level: GzipCompressionLevel,
+}
+
+/// Data structure to pass compression parameters
+#[derive(Debug, Clone, Eq, PartialEq)]
+#[non_exhaustive]
```

Comment:
if this is only `pub(crate)`, it doesn't _really_ need `#[non_exhaustive]`; we only need to put that on public API types.

Reply:
Fixed